openapi-cli/src/test/java/io/ballerina/openapi/cmd/BallerinaCodeGeneratorLicenseTests.java

```diff
@@ -0,0 +1,191 @@
+ */
+public class BallerinaCodeGeneratorLicenseTests extends OpenAPICommandTest {
+
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    @BeforeTest(description = "This will create a new ballerina project for testing below scenarios.")
+    public void setupBallerinaProject() throws IOException {
+        super.setup();
+    }
+
+    @Test(description = "Test openapi to ballerina client generation with default license headers")
+    public void testClientGeneration() throws IOException {
+        Path petstoreYaml = resourceDir.resolve(Paths.get("petstore.yaml"));
+        String[] args = {"--input", petstoreYaml.toString(), "-o", this.tmpDir.toString(), "--mode", "client"};
+        OpenApiCmd cmd = new OpenApiCmd(printStream, tmpDir, false);
+        new CommandLine(cmd).parseArgs(args);
+        cmd.execute();
+
+        if (Files.exists(this.tmpDir.resolve("client.bal")) &&
+                Files.exists(this.tmpDir.resolve("utils.bal")) &&
+                Files.exists(this.tmpDir.resolve("types.bal"))) {
+            try {
+                compareFiles("schema.bal", "types.bal");
+                compareFiles("client.bal", "client.bal");
+                compareFiles("utils_for_client_generation.bal", "utils.bal");
+                deleteGeneratedFiles(false);
+            } catch (IOException e) {
+                Assert.fail(e.getMessage());
+            }
+        } else {
+            Assert.fail("Code generation failed. : " + readOutput(true));
+        }
+    }
+
+    @Test(description = "Test openapi to ballerina service type generation with default license headers")
+    public void testServiceTypeGeneration() throws IOException {
+        Path petstoreYaml = resourceDir.resolve(Paths.get("petstore.yaml"));
+        String[] args = {"--input", petstoreYaml.toString(), "-o", this.tmpDir.toString(), "--with-service-type"};
+        OpenApiCmd cmd = new OpenApiCmd(printStream, tmpDir, false);
+        new CommandLine(cmd).parseArgs(args);
+        cmd.execute();
+
+        if (Files.exists(this.tmpDir.resolve("client.bal")) &&
+                Files.exists(this.tmpDir.resolve("utils.bal")) &&
+                Files.exists(this.tmpDir.resolve("types.bal")) &&
+                Files.exists(this.tmpDir.resolve("service_type.bal"))) {
+            try {
+                compareFiles("schema_for_service.bal", "types.bal");
+                compareFiles("client.bal", "client.bal");
+                compareFiles("utils_for_both_service_client_generation.bal", "utils.bal");
+                compareFiles("service_type.bal", "service_type.bal");
+                deleteGeneratedFiles(false);
+            } catch (IOException e) {
+                Assert.fail(e.getMessage());
+            }
+        } else {
+            Assert.fail("Code generation failed. : " + readOutput(true));
+        }
+    }
+
+    @Test(description = "Test openapi to ballerina generation with default license headers")
```

Comment:
WDYM by "openapi to ballerina generation"? are we talking about the service + client generation?

Reply:
This is for intent to openapi to ballerina code generation common CLI (including service and client).